src/api-binder.ts

```diff
@@ -207,11 +207,16 @@ export class APIBinder {
 		log.debug('Starting current state report');
 		await this.startCurrentStateReport();
 
-		await loadBackupFromMigration(
-			this.deviceState,
-			await this.getTargetState(),
-			bootstrapRetryDelay,
-		);
+		// When we've provisioned, try to load the backup. We
+		// must wait for the provisioning because we need a
+		// target state on which to apply the backup
+		globalEventBus.getInstance().once('targetStateChanged', async state => {
```

Comment:
I'd prefer this as
```suggestion
		globalEventBus.getSingleton().once('targetStateChanged', async state => {
```
because right now it's a bit misleading in thinking there can be multiple instances potentially

Reply:
We're bikeshedding now, and `getInstance` is pretty standard. There's nothing that says this has to be a singleton either, that's just how it's implemented right now.

Reply:
It's not much of a "global" event bus if it's not a singleton :P